extension/js/content_scripts/webmail/gmail-element-replacer.ts

```diff
@@ -783,6 +607,25 @@ export class GmailElementReplacer implements WebmailElementReplacer {
     }
   };
 
+  // loaderEl is a loader reference in case we're processing an attachment
+  // todo: we could also re-use a common method like this in Inbox
+  private handleException = (e: unknown, loaderEl?: JQueryEl) => {
+    if (ApiErr.isAuthErr(e)) {
+      this.notifications.showAuthPopupNeeded(this.acctEmail);
+      loaderEl?.text('Auth needed');
+    } else if (ApiErr.isNetErr(e)) {
+      loaderEl?.text('Network error');
+      // todo:    } else if (ApiErr.isInPrivateMode(e)) {
```

Comment:
Need to figure out where to put this `isInPrivateMode` detection, if still relevant